src/main/java/com/google/api/codegen/metacode/InitCodeNode.java

```diff
@@ -0,0 +1,406 @@
+    typeRef = type;
+    validateType();
+    identifier = table.getNewSymbol(suggestedName);
+    if (children.size() == 0) {
+      if (initValueConfig.hasInitialValue()) {
+        // Update initValueConfig with checked value
+        String newValue = validateValue(type, initValueConfig.getInitialValue());
+        initValueConfig = InitValueConfig.createWithValue(newValue);
+      } else if (initValueConfig.isEmpty()
+          && type.isPrimitive()
+          && !type.isRepeated()
+          && valueGenerator != null) {
+        String newValue = valueGenerator.getAndStoreValue(type, identifier);
+        initValueConfig = InitValueConfig.createWithValue(newValue);
+      }
+    }
+  }
+
+  /**
+   * TODO(michaelbausor): delete this method once DocConfig is no longer used (when python converts
+   * to MVVM)
+   */
+  private void createInitCodeLines() {
+    if (children.size() == 0) {
+      initCodeLine = SimpleInitCodeLine.create(typeRef, identifier, initValueConfig);
+      return;
+    }
+    for (InitCodeNode childItem : children.values()) {
+      childItem.createInitCodeLines();
+    }
+    switch (lineType) {
+      case StructureInitLine:
+        List<FieldSetting> fieldSettings = new ArrayList<>();
+        for (InitCodeNode childItem : children.values()) {
+          FieldSetting fieldSetting =
+              FieldSetting.create(
+                  childItem.typeRef,
+                  childItem.identifier,
+                  childItem.initCodeLine.getIdentifier(),
+                  childItem.initCodeLine.getInitValueConfig());
+          fieldSettings.add(fieldSetting);
+        }
+        initCodeLine = StructureInitCodeLine.create(typeRef, identifier, fieldSettings);
+        break;
+      case ListInitLine:
+        List<Name> elementIdentifiers = new ArrayList<>();
+        for (InitCodeNode childItem : children.values()) {
+          elementIdentifiers.add(childItem.initCodeLine.getIdentifier());
+        }
+        initCodeLine = ListInitCodeLine.create(typeRef, identifier, elementIdentifiers);
+        break;
+      case MapInitLine:
+        Map<String, Name> elementIdentifierMap = new LinkedHashMap<>();
+        for (InitCodeNode childItem : children.values()) {
+          elementIdentifierMap.put(childItem.key, childItem.initCodeLine.getIdentifier());
+        }
+        initCodeLine =
+            MapInitCodeLine.create(
+                typeRef.getMapKeyField().getType(),
+                typeRef.getMapValueField().getType(),
+                typeRef,
+                identifier,
+                elementIdentifierMap);
+        break;
+      default:
+        throw new IllegalArgumentException("Unexpected ParsedNodeType: " + lineType);
+    }
+  }
+
+  /*
+   * Validate the lineType against the typeRef that has been set, and against child objects. In the
+   * case of no child objects being present, update the lineType to SimpleInitLine.
+   */
+  private void validateType() {
```

Comment:
I might prefer this to be static, and have it take lineType and typeRef as arguments